packages/hurl/tests/libcurl.rs

```diff
@@ -753,16 +754,12 @@ fn test_connect_timeout() {
     assert!(matches!(error, HttpError::Libcurl { .. }));
     if let HttpError::Libcurl { code, description } = error {
         eprintln!("description={}", description);
-        if cfg!(target_os = "macos") {
```

Comment:
On Monterey, I've observed that the library returns 28, causing these assertions to fail. However, in GH actions, 7 is returned. This change just asserts more generally for known exit codes and error messages across platforms, since it didn't seem the rest of the app was dependent on the specific error code/message.

Reply:
Hi @jleverenz, thank you for the PR.
I need to test it on Monterey to confirm it. Meanwhile, can you update your PR with the only change ? We prefer to have focused PR and there is some unrelated change in this one (for instance: I don't see if the returned error is due to the updated curl crate, `build_and_test.sh` is changed etc..).
Maybe we can split like this:
- one PR that update lib curl (toml + cargo.lock)
- one PR with the test/libcurl.rs change 
Thanks !

Reply:
Sure thing!

Reply:
Opened a new PR #509 for the second PR. I'll decline this one and open an issue regarding the third issue of OOM errors in GH actions.